packages/freshmint/mint/minters/EditionMinter.ts

```diff
@@ -47,14 +44,17 @@ export type EditionHooks = {
   onComplete: (editionCount: number, nftCount: number) => void;
 };
 
-export class EditionLimitMintError extends FreshmintError {
-  message =
-    'Cannot mint NFTs into an edition with no defined size.\nYour CSV file contains at least one edition with an "edition_size" of null.\nUse the --templates flag to only create the edition template and skip minting.';
+export class InvalidEditionLimitError extends FreshmintError {
+  constructor(value: string, csvLineNumber: number) {
+    super(`Invalid edition on line ${csvLineNumber}: size must be a number, received "${value}".`);
+  }
 }
 
-export class InvalidEditionSizeError extends FreshmintError {
-  constructor(value: string) {
-    super(`Edition size must be a number, received "${value}".`);
+export class ExceededEditionLimitError extends FreshmintError {
+  constructor(edition: Edition) {
+    super(
+      `Invalid edition on line ${edition.csvLineNumber} (with on-chain ID ${edition.id}): target size ${edition.targetSize} exceeds edition limit of ${edition.limit}.`,
```

Comment:
Maybe clarified that the edition limit was set on chain during the first minting and that is non-modifiable.